projects/plugins/jetpack/extensions/blocks/ai-paragraph/edit.js

```diff
@@ -69,13 +68,62 @@ function getSuggestionFromOpenAI(
 		} );
 }
 
+/**
+ * Gather data available in Gutenberg and prepare the best prompt we can come up with.
+ *
+ * @param {Function} select - function returning Gutenberg data store.
+ * @returns {string} - prompt ready to pipe to OpenAI.
+ */
+function preparePromptBasedOnEditorState( select ) {
+	const prompt = formatEditorCanvasContentForCompletion( select( 'core/block-editor' ) );
+
+	// If a user started typing something, we will just create a completion.
+	if ( prompt.length > 0 ) {
+		// We only take the last 240 chars into account, otherwise the prompt gets too long and because we have a 110 tokens limit, there is no place for response.
+		return prompt.slice( -240 );
+	}
+
+	// Let's grab post data so that we can do something smart.
+	const currentPost = select( 'core/editor' ).getCurrentPost();
+	if ( ! currentPost ) {
+		return '';
+	}
```

Comment:
This makes me think we may want to gate the block so it doesn't show in the site or widget editor, where there is no post data?

Reply:
If there is an easy flag than yes, we could prevent it being there. Do you know how to get that?

But I hope to make it more aware of where it is and behave accordingly, as we can also detect the current post type etc.

Reply:
There is no easy solution at the moment I'm afraid, short of checking if the post and post types are defined.

Reply:
When no post is available it will just default to the auto-complete behaviour so that seems fine for now